utils/defaults/defaults.go

```diff
@@ -53,6 +57,22 @@ func GetRevisionHistoryLimitOrDefault(rollout *v1alpha1.Rollout) int32 {
 	return *rollout.Spec.RevisionHistoryLimit
 }
 
+// GetAnalysisRunSucceedHistoryLimitOrDefault returns the specified number of succeed AnalysisRuns to keep or the default number
+func GetAnalysisRunSucceedHistoryLimitOrDefault(rollout *v1alpha1.Rollout) int32 {
+	if rollout.Spec.Analysis == nil || rollout.Spec.Analysis.SucceedRunHistoryLimit == nil {
+		return DefaultAnalysisRunSucceedHistoryLimit
```

Comment:
Add some unit test to pass the `codecov` test?

Reply:
done